src/job.js

```diff
@@ -0,0 +1,110 @@
+import EventObserver from './events';
+import { protobuf, unserialize } from './protobuf';
+import { CYCLE_STATUS_ACCEPTED, CYCLE_STATUS_REJECTED } from './_constants';
+import { GRID_UNKNOWN_CYCLE_STATUS } from './_errors';
+
+export default class Job {
+  constructor({ worker, modelId, modelVersion, gridClient, logger }) {
+    this.worker = worker;
+    this.modelId = modelId;
+    this.modelVersion = modelVersion;
+    this.grid = gridClient;
+    this.plans = {};
+    this.protocols = {};
+    // holds request_key
+    this.cycleParams = {};
+    this.clientConfig = {};
+    this.logger = logger;
+    this.observer = new EventObserver();
+  }
+
+  on(event, handler) {
+    this.observer.subscribe(event, handler);

```

Comment:
Don't do it for any arbitrary `event`. Instead, only do this for `accepted`, `rejected`, and `error`.